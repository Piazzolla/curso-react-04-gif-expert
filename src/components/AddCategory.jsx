import { useState } from "react"

export const AddCategory = ({ onNewCategory }) => {
    const [inputValue, setInputValue] = useState('One Punch');

    const onInputChange = ({ target }) => {
        setInputValue(target.value);
    }

    const onSubmit = (event) => {

        event.preventDefault();
        if (inputValue.trim().length <= 1) return;
        // onAddCategory( categories =>  [ inputValue, ...categories ]);
        onNewCategory(inputValue.trim());
        setInputValue('');

    }
    return (
        <form onSubmit={onSubmit}>
            <input
                type="text"
                placeholder="Buscar gifs"
                value={inputValue}
                onChange={onInputChange}
            />
        </form>
    )
}
