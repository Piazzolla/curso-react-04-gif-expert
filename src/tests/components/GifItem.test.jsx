import { render, screen } from "@testing-library/react"
import { GifItem } from "../../components/GifItem"

describe('Pruebas en GiftItem', () => {
    const title = "Mariano";
    const url = "http://www.google.com/a.jpg";
    test('debe hacer match con el snapshot', () => { 
            const { container } = render( <GifItem title={title} url={url}/>)
            expect( container ).toMatchSnapshot();
    })

    test('debe mostrar la imagen con el URL y el ALT indicado', () => { 
        render( <GifItem title={title} url={url} />);
        
       // screen.debug();// te muestra el sujeto de estudio que trae render
        // expect( screen.getByRole('img').src).toBe( url );
        // expect( screen.getByRole('img').alt).toBe( title );
    //esto mismo mas prolijo:
        const { src, alt } = screen.getByRole('img');
        expect( src ).toBe( url);
        expect( alt ).toBe( title );


    })

    test('debe mostrar el titulo en el componente', () => { 
        render( <GifItem title={title} url={url} />);
        expect( screen.getByText( title )).toBeTruthy(); //me aseguro de que aparte de la property alt este el <p> con el texto
        
     })
})

