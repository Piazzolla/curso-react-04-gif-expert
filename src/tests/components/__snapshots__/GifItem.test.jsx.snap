// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GiftItem debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Mariano"
      src="http://www.google.com/a.jpg"
    />
    <p>
      Mariano
    </p>
  </div>
</div>
`;
